src/v3/pact.ts

```diff
@@ -136,7 +136,10 @@ export class PactV3 {
   }
 
   public withRequest(req: V3Request): PactV3 {
-    this.pact.addRequest(req, req.body && JSON.stringify(req.body))
+    if (typeof req.body !== 'string') {
+      req.body = req.body && JSON.stringify(req.body)
```

Comment:
Thanks for diagnosing and solving this!

I'd prefer not to modify the request object, as it's a parameter passed in to this function. Would you mind updating this function so it doesn't have side effects on the calling object?

Reply:
@TimothyJones good point, I've adjusted the PR